tests/integration/mixer/util.go

```diff
@@ -78,10 +80,13 @@ func VisitProductPage(ing ingress.Instance, timeout time.Duration, wantStatus in
 }
 
 func ValidateMetric(t *testing.T, prometheus prometheus.Instance, query, metricName string, want float64) {
-	got, err := getMetric(t, prometheus, query, metricName)
-	if err != nil {
-		t.Fatalf("%v", err)
-	}
+	var got float64
+	retry.UntilSuccessOrFail(t, func() error {
```

Comment:
Should not do unbounded polls, needs a timeout.

Reply:
There is a default timeout of 30s: https://github.com/istio/istio/blob/506391d42bae03940087fa227b23d310c658c83c/pkg/test/util/retry/retry.go#L33-L36